pa.sh

```diff
@@ -5,6 +5,17 @@ export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/lib/"
 # point to the local downloaded folders
 export PYTHONPATH=${PASH_TOP}/python_pkgs/
 
+## Register the signal handlers, we can add more signals here
+trap kill_all SIGTERM SIGINT
+
+## In case we kill pash via signal, kill all the pending processes
+function kill_all() {
+    # common prefix for all the pash processes
+    pkill -f -9 tmp/pash
```

Comment:
There could be processes that run commands like `grep` and `sort` right? I don't think this kills those.  Also, could this kill the processes of another `pash` instance?

I think that in order to kill all processes we need to do something inside this: https://github.com/binpash/pash/blob/main/runtime/wait_for_output_and_sigpipe_rest.sh in addition to whatever we catch here. 

Also, whenever killing processes, we also need to wait for them.

Also, this should not go to `main` but to `future` since it is an experimental change.

Reply:
It actually kills everything!
In the current state of the script yeah, it will kill other pash instances. However, don't we assume that only a single instance of pash is running on machine?

Reply:
> It actually kills everything!

I don't see why `tmp/pash` would match a `grep` that is left running. Maybe I am missing something though

> In the current state of the script yeah, it will kill other pash instances. However, don't we assume that only a single instance of pash is running on machine?

I don't think this is a valid assumption. We did a lot of modifications to allow multiple instances of pash to be run by the same user.

Reply:
If I am not mistaken, all the pash commands operate on `/tmp/pash_XXXXX/blah`, that's why the  `pkill -f -9 tmp/pash` works for us. 

Is there even a point to optimize the script to only kill the children of the current pash instance?

Reply:
> If I am not mistaken, all the pash commands operate on /tmp/pash_XXXXX/blah, that's why the pkill -f -9 tmp/pash works for us.

A grep in the dataflow might read and write to two completely arbitrary files, therefore I am not sure if `/tmp/pash` would appear anywhere in its invocation.

> Is there even a point to optimize the script to only kill the children of the current pash instance?

This is not an optimizations but rather correctness! If two instances of pash are running we don't want to kill both because one is killed.